ls1intum/Artemis
# `Communication`: Prevent post submission if already in progress

### Checklist

#### General


#### Client

- [ ] I translated all newly inserted strings into English and German. (not applicable)

### Motivation and Context

In communication components, it was possible to repeatedly submit the same message.

* Fixes #6480 

### Description

There are two main changes:

1. The submit button is set to `disabled` while submitting. Changed:
    * `post-create-edit-modal`
    * `anser-post-create-edit-modal`
    * `message-inline-input`
    * `message-reply-inline-input`
2. Before submitting, it is checked if a submission is already in progress
    * This change was made in `PostingCreateEditDirective`, so it affects all components inheriting from this directive.

### Discussion

message-inline-input and message-reply-inline-input are almost identical. From your perspective, would it make sense to make a refactor, which gets rid of the reply input, and the inline input handles both?

### Steps for Testing

1. Log in to Artemis
2. Create a new course or use an existing one
4. Create a new messaging channel or use an existing one
5. Open your browser's dev tools and set network throttling to the slowest connection
6. Type a new message and repeatedly hit either your return key or click the submit button

* The message should be only sent once
* The submit button is in `disabled` state while submitting

#### Exam Mode Testing

1. Log in to Artemis
2. Set up an exam
3. Start the test mode / full exam mode
4. Go to communication / messages
5. Continue w/ steps 5, 6 stated above

#### Performance Review

#### Code Review

#### Manual Tests

#### Exam Mode Test

### Test Coverage

no change in coverage

### Screenshots

![Screen Recording 2023-09-01 at 09 41 08](https://github.com/ls1intum/Artemis/assets/1051396/9f8cfd5a-4f55-4534-bd22-6a7fda171dc3)

Frame rate is a bit low, but you can see the important points
